lib/adapters/http.js

```diff
@@ -15,10 +15,20 @@ var enhanceError = require('../core/enhanceError');
 
 /*eslint consistent-return:0*/
 module.exports = function httpAdapter(config) {
-  return new Promise(function dispatchHttpRequest(resolve, reject) {
+  return new Promise(function dispatchHttpRequest(resolvePromise, rejectPromise) {
+    var timer;
+    var resolve = function resolve(value) {
+      clearTimeout(timer);
+      timer = null;
```

Comment:
I know it isn't necessary but could you also remove the ` && !timer` from the condition below? I think it never applies because the timer is only set inside the `if`.